core/math/transform_2d.cpp

```diff
@@ -164,6 +164,18 @@ Transform2D Transform2D::orthonormalized() const {
 	return ortho;
 }
 
+bool Transform2D::is_conformal() const {
+	// Non-flipped case.
+	if (Math::is_equal_approx(columns[0][0], columns[1][1]) && Math::is_equal_approx(columns[0][1], -columns[1][0])) {
+		return true;
+	}
+	// Flipped case.
+	if (Math::is_equal_approx(columns[0][0], -columns[1][1]) && Math::is_equal_approx(columns[0][1], columns[1][0])) {
+		return true;
+	}
+	return false;
+}
+
```

Comment:
Logic looks good, but it would be great if to avoid branching. Something like
```cpp
return (
    (
        Math::is_equal_approx(columns[0][0], columns[1][1]) &&
        Math::is_equal_approx(columns[0][1], -columns[1][0])
    ) ||
    (
        Math::is_equal_approx(columns[0][0], -columns[1][1]) &&
        Math::is_equal_approx(columns[0][1], columns[1][0])
    )
)
```

Reply:
There is no branching difference, they produce nearly identical code (I was going to say completely identical, but it seems like the asm has superficial differences). https://godbolt.org/z/af6oz9Tsz

Branching happens whenever there are conditions, not just with `if` statements.

Reply:
That's a really cool tool! I'm definitely bookmarking that one :)

Seems like to make it branchless, you have to use `&` instead of `&&` and `|` instead of `||`. I don't know whether getting rid of the branching or allowing for short-circuiting would make it run faster on average.

Reply:
Considering `is_equal_approx` already has branches inside itself, I think it would be better to keep it as-is instead of using `&`. Also it's good for the method to be fast for the common case, which is a valid value, usually a non-flipped one that will hit the first `return true;`.

Reply:
Actually, looking at the code for `is_equal_approx`, it seems like it could also pretty easily be made branchless. Again though, I have no idea whether that would actually make the function faster compared to a short-circuited version. That said, it might benefit from an UNLIKELY statement for the infinity case.

Reply:
That first equal check exists because of infinity, but it still returns true for other values if they are exactly equal, so aside from one branch, this check will be fast for exactly equal values.